UI/CMakeLists.txt

```diff
@@ -108,6 +108,10 @@ foreach(graphics_library IN ITEMS opengl metal d3d11)
   endif()
 endforeach()
 
+get_property(obs_module_list GLOBAL PROPERTY OBS_MODULES_ENABLED)
+list(JOIN obs_module_list "|" SAFE_MODULES)
```

Comment:
I'd prefer using `configure_file` to output a source file `obs_safe_modules.c` that sets up a static array directly and also contains utility functions for that list. Gives us encapsulation, a single location to manage basic functionality, and doesn't rely on macros/preprocessor strings.

I would expect the contents of this file to be quite static, so it shouldn't hurt having it added directly as a compilation unit. If the functionality is used in UI code only, we could generate it as C++ directly even.

Reply:
Originally this used a generated header though based on review comments I changed it to a define (https://github.com/obsproject/obs-studio/pull/8455#discussion_r1134005599).

Reply:
Yeah I’d have CMake generate straight C code (so no defines), but we can also change that in bulk at a later time, similar to the version library.

Reply:
Originally I created a header file containing a `unordered_set<string>` with the module names: https://github.com/obsproject/obs-studio/commit/7031ee85c61c33c04f3ae0b1be05dae6d728b9ac#diff-5d2ded8a0a8649fab4ca9479064e8938310c4d65522dea409764ec6b776ea25c

If that's what you mean I can simply go back to that. Otherwise yes we can do it later.

Reply:
Damn that’s kinda what I meant, probably brainlagged in the past. But let’s go with the current implementation to get this over the finish line.